### PR TITLE
Enable to build and install host-verify package and run host-verify sample on CI/CD

### PR DESCRIPTION
**Pull request by ryanhsu19**
Enable testing of `host-verify` package and sample on `Azure-Linux` pipeline of CI/CD. 

After SGX certificates and an SGX report are generated, they will be copied to a non-SGX Ubuntu Linux machine and a non-SGX Windows machine.

On a non-SGX machine, regardless of the operating system installed, it will build a `open-enclave-hostverify` package, install it, and run the `host-verify` sample. ~Note that package installation is not included in this pull request.~

Signed-off-by: Ryan Hsu <ryhsu@microsoft.com>

### CODE REVIEW

**Comment by ryanhsu19** [comment]
bors try

File: .jenkins/pipelines/Azure/Linux/Jenkinsfile [hunk 1]
```diff
@@ -229,6 +229,17 @@ def ACCHostVerificationTest(String version, String build_type) {
                            cmake ${WORKSPACE} -G Ninja -DBUILD_ENCLAVES=OFF -DHAS_QUOTE_PROVIDER=OFF -DCMAKE_BUILD_TYPE=${build_type} -Wdev
                            ninja -v
                            ctest -R host_verify --output-on-failure --timeout ${CTEST_TIMEOUT_SECONDS}
+                           cp tests/host_verify/host/*.der ${WORKSPACE}/samples/host_verify
+                           cp tests/host_verify/host/*.bin ${WORKSPACE}/samples/host_verify
+                           pushd ${WORKSPACE}/samples/host_verify
+                           mkdir build
+                           cd build
+                           cmake .. -G Ninja -DBUILD_ENCLAVES=OFF -DHAS_QUOTE_PROVIDER=OFF -DCMAKE_BUILD_TYPE=${build_type} -Wdev
```
Comment (anakrish) [inline .jenkins/pipelines/Azure/Linux/Jenkinsfile:237]: You can remove the `-DHAS_QUOTE_PROVIDER=OFF`. That option is no longer valid.

Comment (ryanhsu19) [reply .jenkins/pipelines/Azure/Linux/Jenkinsfile:237]: In both cases?

File: .jenkins/pipelines/Azure/Linux/Jenkinsfile [hunk 2]
```diff
@@ -229,6 +229,17 @@ def ACCHostVerificationTest(String version, String build_type) {
...
+                           ninja -v
+                           ./host_verify -r ../sgx_report.bin
+                           ./host_verify -c ../sgx_cert_rsa.der
+                           cd ..
```
Comment (yentsanglee) [inline .jenkins/pipelines/Azure/Linux/Jenkinsfile:241]: You can test sgx_cert_ec.der too. Same below.

Comment (yentsanglee) [inline .jenkins/pipelines/Azure/Linux/Jenkinsfile:241]: "cd .." is not needed before "popd". Same below.

**Comment by ryanhsu19** [comment]
bors try

**Comment by ryanhsu19** [comment]
bors try

**Comment by ryanhsu19** [comment]
Weird, bors did not respond to me. Let me try again.
bors try

**Comment by ryanhsu19** [comment]
bors try

**Comment by ryanhsu19** [comment]
Set up CMAKE_PREFIX_PATH in order to resolve it.
bors try

File: .jenkins/pipelines/Azure/Linux/Jenkinsfile (hunk 1 above)
Comment (BRMcLaren) [reply .jenkins/pipelines/Azure/Linux/Jenkinsfile]: In all cases, yes.

File: .jenkins/pipelines/Azure/Linux/Jenkinsfile [hunk 3]
```diff
@@ -248,7 +260,19 @@ def ACCHostVerificationTest(String version, String build_type) {
                         vcvars64.bat x64 && \
                         cmake.exe ${WORKSPACE} -G Ninja -DBUILD_ENCLAVES=OFF -DHAS_QUOTE_PROVIDER=OFF -DCMAKE_BUILD_TYPE=${build_type} -DNUGET_PACKAGE_PATH=C:/oe_prereqs -Wdev && \
                         ninja -v && \
-                        ctest.exe -V -C ${build_type} -R host_verify --output-on-failure --timeout ${CTEST_TIMEOUT_SECONDS}
+                        ninja install && \
```
Comment (BRMcLaren) [inline .jenkins/pipelines/Azure/Linux/Jenkinsfile:263]: From my understanding, this test is to verify the quotes created at the earlier steps. This is now modified to then also install the host verification package and install the samples and test.

I think this should stay separate and we should have a HostVerificationPackage testing phase instead of appending to the end of an existing (long-running) test.

Comment (yentsanglee) [reply .jenkins/pipelines/Azure/Linux/Jenkinsfile:263]: @BRMcLaren are you suggesting we wait until #3271 is resolved and we can pick a package from build?

Comment (BRMcLaren) [reply .jenkins/pipelines/Azure/Linux/Jenkinsfile:263]: I'm suggesting make another test that specifically builds the package, installs it and the runs the samples. It would likely be better than tacking it on to the end of a current test.

The cpack issue you linked is unrelated to host verify, it's for the sdk package.

**Comment by ryanhsu19** [comment]
bors try

**Comment by ryanhsu19** [comment]
bors try

**Comment by ryanhsu19** [comment]
Weird, the directory "." works fine on my work computer and Windows VMs but not on Jenkins. Now trying again.
bors try

File: .jenkins/pipelines/Azure/Linux/Jenkinsfile (hunk 3 above)
Comment (ryanhsu19) [reply .jenkins/pipelines/Azure/Linux/Jenkinsfile]: Done with the new function, I would like both of you to review it again, please. Thanks!

File: .jenkins/pipelines/Azure/Linux/Jenkinsfile [hunk 4]
```diff
@@ -269,6 +401,9 @@ try{
         "Host verification 1604 Release":         { ACCHostVerificationTest('16.04', 'Release') },
         "Host verification 1804 Release":         { ACCHostVerificationTest('18.04', 'Release') },
 
+        "Host verification package 1604 Release": { ACCHostVerificationPackageTest('16.04', 'Release') },
```
Comment (BRMcLaren) [inline .jenkins/pipelines/Azure/Linux/Jenkinsfile:404]: Please add "RelWithDebInfo" to the test matrix, we release Linux packages in this mode.

File: .jenkins/pipelines/Azure/Linux/Jenkinsfile [hunk 5]
```diff
@@ -256,6 +256,138 @@ def ACCHostVerificationTest(String version, String build_type) {
...
+    /* Linux nonSGX stage. */
+    stage("Linux nonSGX Verify Quote") {
+        node(AGENTS_LABELS["ubuntu-nonsgx"]) {
+            timeout(GLOBAL_TIMEOUT_MINUTES) {
+                cleanWs()
+                checkout scm
+                unstash "linux_host_verify-${version}-${build_type}-${BUILD_NUMBER}"
+                def task = """
+                           cmake ${WORKSPACE} \
+                             -G Ninja \
```
Comment (BRMcLaren) [inline .jenkins/pipelines/Azure/Linux/Jenkinsfile:318]: We are ultimately building the host verification package twice, could we change one of these to build without ninja and one with?  This way we cover an edge case of it working with ninja and not in cpack.

File: .jenkins/pipelines/Azure/Linux/Jenkinsfile [hunk 6]
```diff
@@ -256,6 +256,138 @@ def ACCHostVerificationTest(String version, String build_type) {
...
+    */
+    stage("ACC-1804 Generate Quote") {
+        node(AGENTS_LABELS["acc-ubuntu-18.04"]) {
+            timeout(GLOBAL_TIMEOUT_MINUTES) {
+                cleanWs()
+                checkout scm
+
+                println("Generating certificates and reports ...")
+                def task = """
+                           cmake ${WORKSPACE} -G Ninja -DHAS_QUOTE_PROVIDER=ON -DCMAKE_BUILD_TYPE=${build_type} -Wdev
```
Comment (BRMcLaren) [inline .jenkins/pipelines/Azure/Linux/Jenkinsfile:271]: -DHAS_QUOTE_PROVIDER

is deprecated and should not be used.

File: .jenkins/pipelines/Azure/Linux/Jenkinsfile (hunk 5 above)
Comment (ryanhsu19) [reply .jenkins/pipelines/Azure/Linux/Jenkinsfile]: Got it. I replaced `ninja` with `make` for the Linux non-SGX node.

**Comment by ryanhsu19** [comment]
bors try

File: .jenkins/pipelines/Azure/Linux/Jenkinsfile [hunk 7]
```diff
@@ -256,6 +256,137 @@ def ACCHostVerificationTest(String version, String build_type) {
...
+
+    /* Linux nonSGX stage. */
+    stage("Linux nonSGX Verify Quote") {
+        node(AGENTS_LABELS["ubuntu-nonsgx"]) {
+            timeout(GLOBAL_TIMEOUT_MINUTES) {
+                cleanWs()
+                checkout scm
+                unstash "linux_host_verify-${version}-${build_type}-${BUILD_NUMBER}"
+                def task = """
+                           cmake ${WORKSPACE} \
```
Comment (anakrish) [inline .jenkins/pipelines/Azure/Linux/Jenkinsfile:317]: Instead of building the SDK, is it possible to just stach the `host_verify` executable as well and then unstash it here and test it?

Comment (ryanhsu19) [reply .jenkins/pipelines/Azure/Linux/Jenkinsfile:317]: We'd like to ensure `host_verify` can be built after the package installation, so that's why we do it after the package installation instead of stashing the executable and unstashing it on another machine.

**Comment by radhikaj** [comment]
@nonpolarity Can you comment on this change please?
